pennylane/operation.py

```diff
@@ -228,10 +228,42 @@ def expand_matrix(base_matrix, wires, wire_order):
 # =============================================================================
 
 
-class NoDecompositionError(Exception):
-    """Raised when an Operator does not have a decomposition defined."""
+class OperatorPropertyUndefined(Exception):
+    """Generic exception to be used for undefined
+    Operator properties or methods."""
+
+
+class DecompositionUndefinedError(OperatorPropertyUndefined):
+    """Raised when an Operator's representation as a decomposition is undefined."""
+
+
+class TermsUndefinedError(OperatorPropertyUndefined):
+    """Raised when an Operator's representation as a linear combination is undefined."""
+
+
+class MatrixUndefinedError(OperatorPropertyUndefined):
+    """Raised when an Operator's matrix representation is undefined."""
+
+
+class SparseMatrixUndefinedError(OperatorPropertyUndefined):
+    """Raised when an Operator's sparse matrix representation is undefined."""
 
-    pass
```

Comment:
Christina had a pass, but I assume errors don't need that?

Reply:
Having a docstring is an implicit pass, for all classes and functions (and I think preferred by PEP8) 🙂